### PR TITLE
KBV-360 Inline the IssueCredentialFunctionRole

### PR DESCRIPTION
**Pull request by idrop**
## Proposed changes

* Remove the IssueCredentialFunctionRole and map the policy on the IssueCredentialFunction directly
* tidy up KMS sign and SSM policies

### Issue tracking

- [KBV-360](https://govukverify.atlassian.net/browse/KBV-360)

### CODE REVIEW

File: infrastructure/lambda/template.yaml [hunk 1]
```diff
@@ -346,66 +346,6 @@ Resources:
               Resource:
                 - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${AWS::StackName}/clients/*"
 
-  IssueCredentialFunctionRole:
```
Comment (idrop) [inline infrastructure/lambda/template.yaml:349]: This role is now inline in the IssueCredentialFunction, like the other lambdas

File: infrastructure/lambda/template.yaml [hunk 2]
```diff
@@ -460,13 +422,6 @@ Resources:
             Action:
               - 'kms:*'
             Resource: '*'
-          - Sid: "Allow use of the key for digital signing"
-            Effect: Allow
-            Principal:
```
Comment (idrop) [inline infrastructure/lambda/template.yaml:465]: This is now set on the IssueCredentialFunction

File: infrastructure/lambda/template.yaml [hunk 3]
```diff
@@ -737,13 +692,6 @@ Resources:
       Value: !FindInMap [OrdnanceSurveyAPIURLMapping, Environment, !Ref 'Environment']
       Description: Ordnance Survey Postcode Lookup API URL
 
-  IssueCredentialFunctionPermission:
-    Type: AWS::Lambda::Permission
-    Properties:
-      Action: lambda:InvokeFunction
-      FunctionName: !GetAtt IssueCredentialFunction.Arn
-      Principal: apigateway.amazonaws.com
```
Comment (idrop) [inline infrastructure/lambda/template.yaml:745]: moved the permission to where the other permissions are in the template

File: infrastructure/lambda/template.yaml [hunk 4]
```diff
@@ -415,7 +355,29 @@ Resources:
...
+            TableName: !Ref AddressTable
+        - Statement:
+            Effect: Allow
+            Action:
+              - "kms:Sign"
+            Resource: !Sub "arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/${AddressCriVcSigningKey}"
+        - SQSSendMessagePolicy:
+            QueueName: !ImportValue AuditEventQueueName
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub "${AWS::StackName}/*"
```
Comment (idrop) [inline infrastructure/lambda/template.yaml:373]: What do you think of this? These params aren't secret, and I can't think of a securty reason that a lambda shouldn't have access to read these stack-scoped params. If we adopt this on other lambdas it'll reduce boilerplate yaml

Comment (KaraokeStu) [reply infrastructure/lambda/template.yaml:373]: Any parameters that specifically need locking down should be able to be overridden by IAM so I think we're good, plus we're limiting to our own stack - so we're good there too

File: infrastructure/lambda/template.yaml [hunk 5]
```diff
@@ -460,13 +422,6 @@ Resources:
             Action:
```
Comment (KaraokeStu) [inline infrastructure/lambda/template.yaml:422]: Do we really want to be granting access to ALL of KMS functionality for ALL resources (regardless of region, stack, etc?)

I think this should at least be locked down to the stack or at very least environment

**Review by jkunle — APPROVED** [review]
We should investigate further the KMS security

**Pull request merged**